Fix #1548 by providing a new desktop.HiddenCursor cursor for invisible cursor

MagicalTux opened the pull request:
### Description:
Add a new `desktop.InvisibleCursor` cursor that can be returned by widgets implementing the `Cursorable` interface.

Fixes #1548 

### Checklist:
<!-- Please tick these as appropriate using [x] -->

- [ ] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

andydotxyz commented:
A neat approach, well done.
I wonder if the naming might be ajusted - we have not used the word "Invisible" before - maybe "Hidden" is more consistent but still conveys the same outcome?

MagicalTux commented:
Merge request updated:

* Renamed `InvisibleCursor` to `HiddenCursor` since this is now what this is
* Added a check so we only call SetCursor() if cursor has changed

andydotxyz reviewed (approved):
Looks good to me, thanks. I would like to get @stuartmscott or @toaster feedback on this as well if possible.

stuartmscott reviewed (approved):
HiddenCursor is a better name, though the PR title is now out of date. Otherwise, looks good to me!

andydotxyz commented:
> though the PR title is now out of date

Fixed ;)
